src/main/java/com/streamr/client/authentication/AuthenticationMethod.java

```diff
@@ -8,20 +8,24 @@
 import okhttp3.Request;
 import okhttp3.RequestBody;
 import okhttp3.Response;
+import okio.BufferedSource;
 
 import java.io.IOException;
 import java.util.Date;
 
 public abstract class AuthenticationMethod {
 
-    protected final StreamrClientOptions options;
-    private JsonAdapter<LoginResponse> responseAdapter;
+    private String restApiUrl;
+    private JsonAdapter<LoginResponse> responseAdapter = HttpUtils.MOSHI.adapter(LoginResponse.class);
 
-    public AuthenticationMethod(StreamrClientOptions options) {
-        this.options = options;
+    public AuthenticationMethod() {
```

Comment:
Now there's no way to use this class without explicitly calling `setRestApiUrl`. It is required, so to enforce that we have it when we need it, we have two options:
- pass it to the constructor (difficult for the user)
- pass it to the public methods: `login(String restApiUrl)` (perhaps better, as the trouble of doing so will remain internal to our code?)